Fix the path of embed source maps

IgorSalevan opened the pull request:
Related to: https://oat-sa.atlassian.net/browse/TDR-7

Added a fix which prepends CSS source-maps with their relative paths.

Companion PR:
[oat-sa/extension-tao-itemqti/: #1366](https://github.com/oat-sa/extension-tao-itemqti/pull/1366)

IgorSalevan committed:
TDR-7: fix: prepending css sourseMap resources with the relative paths.

IgorSalevan committed:
bumped version to 0.3.2

IgorSalevan committed:
TDR-7: Assigned default value (object) to the module parameter `paths`

IgorSalevan committed:
TDR-7: made regExp for extractiong 'css!' imports wider in order to cover cases when extension (css) is not set for the imported files.

krampstudio commented on lib/sourceMapPathResolve.js:
```diff
@@ -0,0 +1,130 @@
+ * @param {String} source - file contents
+ * @returns {Array<{file: String, path: String}>} matches - an array of found sourceMap files
+ */
+const matchAllSourceMaps = source => {
```
The rule is to use arrow function in the following situations : 
 - increase readability (one line) 
 - prevent a closure or binding the lexical scope 

So in this case, we prefer using plain function : 
 ```
function matchAllSourceMaps(source) {
```

IgorSalevan replied:
applied your recommendation about the arrow funcs.

krampstudio commented on lib/transformer.js:
```diff
@@ -51,6 +56,51 @@ module.exports = async function transformer({
+    //     }
+    //     return matches;
+    // };
+
```
You forgot to remove that part

krampstudio commented on lib/sourceMapPathResolve.js:
```diff
@@ -0,0 +1,130 @@
+ *
+ * @param {String} mapBody
+ * @param {String} codeBody
+ * @param {Object} paths
```
can you explain what are those paths ?

IgorSalevan replied:
added a comment to the `paths` param

krampstudio commented on lib/sourceMapPathResolve.js:
```diff
@@ -0,0 +1,130 @@
+        code: codeBody
+    };
+
+    for (const i in matchedSourceMaps) {
```
If `matchedSourceMaps` is an array I prefer the iteration to be made using `for ... of ` or `.forEach` (since for ... in doesn't guarantee the index order)

IgorSalevan replied:
Yes I know that index order maybe not preserved. But changed the implementation and used forEach and got rid of the index.

krampstudio commented on lib/sourceMapPathResolve.js:
```diff
@@ -0,0 +1,130 @@
+
+    for (const i in matchedSourceMaps) {
+        // Extract the sourceMap path from the `css!` imports
+        const matchResult = (new RegExp(`css!([\\w|\\-|.|\\/]+)/${matchedSourceMaps[i].file}([\\.css|\\'\\"])`))
```
why do you need the parenthesis around the RegExp ?

IgorSalevan replied:
Just to surround the dynamic creation of object which is chained then with its method.
Removed.

krampstudio reviewed (changes requested):
That's good but can you please  : 
  - add a test for the resolver ? 
 - give detailed test instructions 
 - the sourcemap path is replaced by an alias instead of the path in some situation, we should resolve them (the matching alias/path is available at https://github.com/oat-sa/grunt-tao-bundle/blob/master/lib/bundler.js#L58)

IgorSalevan committed:
TDR-7: fixes, based on PR review comments.

IgorSalevan committed:
TDR-7:
- fixed regular expression
- applied sourceMapResolver to all types of bundle (babel, uglify,...)

IgorSalevan committed:
TDR-7: added tests for the sourceMapPathResolve module

krampstudio commented on lib/sourceMapPathResolve.js:
```diff
@@ -0,0 +1,134 @@
+                if (extractedPath.indexOf(pathKey) === 0) {
+                    if (matchedSourceMap.file === 'waitingDialog') {
+                        console.log(extractedPath, pathKey, pathValue);
+                    }
```
can you please remove this block

IgorSalevan replied:
Done.

krampstudio commented:
can you also add a line in the README release history

IgorSalevan committed:
Removed debug code.

IgorSalevan committed:
Updated the release history.

IgorSalevan committed:
Merge remote-tracking branch 'origin/develop' into fix/TDR-7/paths-of-embeded-source-maps

# Conflicts:
#	README.md
#	package-lock.json
#	package.json

krampstudio committed:
update versions and history
